uptane/services/inventorydb.py

```diff
@@ -84,122 +84,94 @@ def scrub_filename(fname, expected_containing_dir):
-  def save_vehicle_manifest(self, vin, signed_vehicle_manifest):
-    """
-    Given a manifest of form
-    uptane.formats.SIGNABLE_VEHICLE_VERSION_MANIFEST_SCHEMA, save it in an index
-    by vin, and save the individual ecu attestations in an index by ecu serial.
-    """
-    vehicle_primary_ids = self.inventory_db['public_keys']['vehicle_primaries']
-    ecu_serial_ids = self.inventory_db['public_keys']['all_ecus']
-    if len(vehicle_primary_ids) == 0 or len(ecu_serial_ids) == 0:
-      raise uptane.UnknownECU("InventoryDataBase has not registered yet!")
-
-    vehicle_ids = self.inventory_db['vehicle_manifests']
-    if (vehicle_ids and (vin in vehicle_ids)):
-      (self.inventory_db['vehicle_manifests'][vin]).append(signed_vehicle_manifest)
-    else:
-      vm_list = []
-      vm_list.append(signed_vehicle_manifest)
-      vehicle_ids[vin] = vm_list
-      self.inventory_db['vehicle_manifests'] = vehicle_ids
-
-
-  # Get vehicle manifest
-  def get_vehicle_manifest(self, vin):
-    # Check arg format
-    vehicle_ids = self.inventory_db['vehicle_manifests']
-    if (vehicle_ids and (vin in vehicle_ids)):
-      return self.inventory_db['vehicle_manifests'][vin]
-    return False
-
-
-  # Save ECU manifest
-  def save_ecu_manifest(self, ecu_serial, signed_ecu_manifest):
-    uptane.formats.VIN_SCHEMA.check_match(vin)
-    uptane.formats.ECU_SERIAL_SCHEMA.check_match(ecu_serial)
-    uptane.formats.SIGNABLE_ECU_VERSION_MANIFEST_SCHEMA.check_match(
-        signed_ecu_manifest)
-
-    ecu_serial_ids = self.inventory_db['ecu_manifests']
-    if (ecu_serials_ids and (ecu_serial in ecu_serials_ids)):
-      (self.inventory_db['ecu_manifests'][ecu_serial]).append(signed_vehicle_manifest)
-    else:
-      ecu_list = []
-      ecu_list.append(signed_vehicle_manifest)
-      ecu_serial_ids[ecu_serial] = ecu_list
-      self.inventory_db['ecu_manifests'] = ecu_serial_ids
 
+# Global dictionaries
+vehicle_manifests_dic = {'vehicle_manifests': {}}
+ecu_manifests_dic = {'ecu_manifests': {}}
+primary_public_keys = {'vehicle_primaries': {}}
+public_keys = {'all_ecus': {}}
+
+
+
+# Save ECU manifest
+def save_ecu_manifest(ecu_serial, signed_ecu_manifest):
+
+  uptane.formats.VIN_SCHEMA.check_match(vin)
+  uptane.formats.ECU_SERIAL_SCHEMA.check_match(ecu_serial)
+  uptane.formats.SIGNABLE_ECU_VERSION_MANIFEST_SCHEMA.check_match(
+       signed_ecu_manifest)
 
-  # Get ECU manifest
-  def get_ecu_manifest(self, ecu_serial):
-    # Check arg format
-    ecu_serial_ids = self.inventory_db['ecu_manifests']
-    if (ecu_serial_ids and (ecu_serial in ecu_serial__ids)):
-      return self.inventory_db['ecu_manifests'][ecu_serial]
-    return False
+  if ecu_serial not in public_keys['all_ecus']:
+    raise uptane.UnknownECU("ECU serial has not been registered yet!")
+
+  if ecu_serial not in ecu_manifests_dic['vehicle_manifests']:
+    ecu_list = []
+    ecu_list.append(signed_ecu_manifest)
+    ecu_manifests_dic['ecu_manifests'][ecu_serial] = ecu_list
```

Comment:
I think I already commented on this.

The previous three lines should just be:
ecu_manifests_dic['ecu_manifests'][ecu_serial] = [signed_ecu_manifest]

There's no need for the creation of a temp empty list and a separate line to append it to an empty list.